eva/executor/create_index_executor.py

```diff
@@ -56,40 +55,32 @@ def exec(self):
                 col for col in feat_catalog_entry.columns if col.name == feat_col_name
             ][0]
 
+            # Get udf function.
+            udf_func = self.node.udf_func
+
             # Add features to index.
             # TODO: batch size is hardcoded for now.
             index = None
-            input_dim, num_rows = -1, 0
-            logical_to_row_id = dict()
+            input_dim = -1
             storage_engine = StorageEngine.factory(feat_catalog_entry)
-            for batch in storage_engine.read(feat_catalog_entry, 1):
-                # Pandas wraps numpy array as an object inside a numpy
-                # array. Use zero index to get the actual numpy array.
-                feat = batch.column_as_numpy_array(feat_col_name)[0]
+            for input_batch in storage_engine.read(feat_catalog_entry, 1):
+                if udf_func:
```

Comment:
We should add some comments on what is happening -- like SimilaritySearch(f).x -> f